inference/enterprise/workflows/execution_engine/introspection/blocks_loader.py

```diff
@@ -0,0 +1,249 @@
+    try:
+        return _load_blocks_from_plugin(plugin_name=plugin_name)
+    except ImportError as e:
+        raise PluginLoadingError(
+            public_message=f"It is not possible to load workflow plugin `{plugin_name}`. "
+            f"Make sure the library providing custom step is correctly installed in Python environment.",
+            context="workflow_compilation | blocks_loading",
+            inner_error=e,
+        ) from e
+    except AttributeError as e:
+        raise PluginInterfaceError(
+            public_message=f"Provided workflow plugin `{plugin_name}` do not implement blocks loading "
+            f"interface correctly and cannot be loaded.",
+            context="workflow_compilation | blocks_loading",
+            inner_error=e,
+        ) from e
+
+
+def _load_blocks_from_plugin(plugin_name: str) -> List[BlockSpecification]:
+    module = importlib.import_module(plugin_name)
+    blocks = module.load_blocks()
+    return [
+        BlockSpecification(
+            block_source=plugin_name,
+            identifier=get_full_type_name(selected_type=block),
+            block_class=block,
+            manifest_class=block.get_input_manifest(),
+        )
+        for block in blocks
+    ]
+
+
+def load_initializers() -> Dict[str, Union[Any, Callable[[None], Any]]]:
+    plugins_to_load = os.getenv(WORKFLOWS_PLUGINS_ENV)
+    if plugins_to_load is None:
+        return {}
+    result = {}
+    for plugin_name in plugins_to_load.split(","):
+        result.update(load_initializers_from_plugin(plugin_name=plugin_name))
+    return result
+
+
+def load_initializers_from_plugin(
+    plugin_name: str,
+) -> Dict[str, Union[Any, Callable[[None], Any]]]:
+    try:
+        logging.info(f"Loading workflows initializers from plugin {plugin_name}")
+        return _load_initializers_from_plugin(plugin_name=plugin_name)
+    except ImportError as e:
+        raise PluginLoadingError(
+            public_message=f"It is not possible to load workflow plugin `{plugin_name}`. "
+            f"Make sure the library providing custom step is correctly installed in Python environment.",
+            context="workflow_compilation | blocks_loading",
+            inner_error=e,
+        ) from e
+
+
+def _load_initializers_from_plugin(
+    plugin_name: str,
+) -> Dict[str, Callable[[None], Any]]:
+    module = importlib.import_module(plugin_name)
+    registered_initializers = getattr(module, "REGISTERED_INITIALIZERS", {})
+    return {
+        f"{plugin_name}.{parameter_name}": initializer
+        for parameter_name, initializer in registered_initializers.items()
+    }
+
+
+def _validate_loaded_blocks_names_uniqueness(blocks: List[BlockDescription]) -> None:
+    block_names_lookup = {}
+    for block in blocks:
+        if block.human_friendly_block_name in block_names_lookup:
+            clashing_block = block_names_lookup[block.human_friendly_block_name]
+            raise PluginLoadingError(
```

Comment:
Maybe we should just take the first one instead (with priority order: user-added, then core).

We may eventually need a way to namespace community ones. Maybe we'd have users prefix them with their workspace like `some-user/block_name`? 

Apple does FQDN (`com.roboflow.workflows.block_name`) but that's always seemed really weird to me since you don't actually have to own that domain name to use it in your FQDN and the "subdomains" don't correlate with anything external afaict.